Userland/Applications/PackageManager/AvailablePort.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Stream.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/EventLoop.h>
+#include <LibCore/File.h>
+#include <LibCore/Proxy.h>
+#include <LibCore/System.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibHTTP/HttpResponse.h>
+#include <LibMain/Main.h>
+#include <LibMarkdown/Document.h>
+#include <LibMarkdown/Table.h>
+#include <LibProtocol/Request.h>
+#include <LibProtocol/RequestClient.h>
+
+#include <Applications/PackageManager/AvailablePort.h>
+#include <Applications/PackageManager/MarkdownTableFinder.h>
+
+#include <Shell/AST.h>
+#include <Shell/Formatter.h>
+#include <Shell/NodeVisitor.h>
+#include <Shell/PosixParser.h>
+#include <Shell/Shell.h>
+
+static bool is_installed(Vector<InstalledPort>& installed_ports_database, StringView package_name)
+{
+    for (auto& port : installed_ports_database) {
+        if (port.name() == package_name)
+            return true;
+    }
+    return false;
```

Comment:
Looks like we might want a `HashMap<String, InstalledPort> installed_ports_database` instead? It's most likely not performance critical, given the number of ports, but there is no real reason against it either (at least as far as I can tell from a glance).